src/App/Datums.cpp

```diff
@@ -21,42 +23,318 @@
  ***************************************************************************/
 
 #include "PreCompiled.h"
+#ifndef _PreComp_
+#include <string>
+#endif
+
+#include <App/Document.h>
+#include <Base/Exception.h>
+#include <Base/Placement.h>
 
 #include "Datums.h"
-#include "Document.h"
-#include "Origin.h"
 
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
 
 using namespace App;
 
 PROPERTY_SOURCE(App::DatumElement, App::GeoFeature)
 PROPERTY_SOURCE(App::Plane, App::DatumElement)
 PROPERTY_SOURCE(App::Line, App::DatumElement)
+PROPERTY_SOURCE(App::Point, App::DatumElement)
+PROPERTY_SOURCE(App::LocalCoordinateSystem, App::GeoFeature)
 
-DatumElement::DatumElement()
+DatumElement::DatumElement(bool hideRole)
```

Comment:
Boolean blindness

Reply:
Do we really need to change this? It's a single argument it's not like it's problematic. I don't really understand this boolean blindness anyway. Unless someone is editing source code with notepad, the IDE let you see the signature of the function easily enough.

I think rejecting any use of boolean is a bit drastic. Searching for "(bool " in the source code gives 1510 results. If for each of those function we implemented a enum class, imagine the mess that would be.

If you still want me to take action please advise your prefered replacement solution. Thanks

Reply:
@PaddleStroke, when editing your commits, please run `clang-format -i src/App/whatever.cpp` for files in App directory to be sure they still conform.
(as someone who is editing code with Vim, I'm not big fan of declaring enum for sole purpose of boolean blindness idiom)

Reply:
All of these review suggestions are that: suggestions. @wwmayer first introduced me to the modern notion of "boolean blindness", and at the call site I do find it is easier to read calls that don't have booleans in them. With just a single argument the benefit of switching to an enumeration is marginal (at best), but I am happy to defer to @wwmayer here about whether in these cases we should indeed require a new enumeration.

Reply:
After thinking about it, I think this boolean can be removed and replaced by a function 'hideRole'. Then this function can be called after the initialization.
However as this boolean is not used in this PR (it will be in next PR implementing the Part subclasses of those datums) I ask that we leave this as it is now as to not create more conflicts. And I'll replace this bool in this next PR.